client/settings/use-payment-method-availability/index.tsx

```diff
@@ -0,0 +1,214 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import methodsConfiguration from 'wcpay/payment-methods-map';
+import { upeCapabilityStatuses } from 'wcpay/settings/constants';
+import {
+	useEnabledPaymentMethodIds,
+	useGetPaymentMethodStatuses,
+	useManualCapture,
+} from 'wcpay/data';
+import PAYMENT_METHOD_IDS from 'wcpay/constants/payment-method';
+import { getMissingCurrenciesTooltipMessage } from 'multi-currency/utils/missing-currencies-message';
+import { __, sprintf } from '@wordpress/i18n';
+import interpolateComponents from '@automattic/interpolate-components';
+import { ExternalLink } from '@wordpress/components';
+import { ChipType } from 'wcpay/components/chip';
+
+const documentationTypeMap = {
+	DEFAULT:
+		'https://woocommerce.com/document/woopayments/payment-methods/additional-payment-methods/#method-cant-be-enabled',
+	BNPLS:
+		'https://woocommerce.com/document/woopayments/payment-methods/buy-now-pay-later/#contact-support',
+};
+
+const getDocumentationUrlForDisabledPaymentMethod = (
+	paymentMethodId: string,
+	isPoInProgress = false
+): string => {
+	if ( isPoInProgress ) {
+		return 'https://woocommerce.com/document/woopayments/startup-guide/gradual-signup/#additional-payment-methods';
+	}
+	if ( methodsConfiguration?.[ paymentMethodId ]?.allows_pay_later ) {
+		return documentationTypeMap.BNPLS;
+	}
+
+	return documentationTypeMap.DEFAULT;
+};
+
+const usePaymentMethodAvailability = ( id: string ) => {
```

Comment:
Would you mind adding a docblock describing the main intention/functionality of the hook please?

Reply:
Sure, done in 02ea6aaa5

Reply:
Perfect. Thanks!